server/src/main/java/io/crate/role/Role.java

```diff
@@ -53,18 +58,29 @@ public class Role implements Writeable, ToXContent {
         "crate",
         new RolePrivileges(Set.of()),
         Set.of(),
-        new Properties(true, null, null),
+        new Properties(true, null, null, null),
         true);
 
     public record Properties(boolean login,
                              @Nullable SecureHash password,
-                             @Nullable JwtProperties jwtProperties) implements Writeable, ToXContent {
+                             @Nullable JwtProperties jwtProperties,
+                             Map<String, Object> sessionSettings) implements Writeable, ToXContent {
```

Comment:
Can we use `SessionSettings` here instead of Map  in order to re-use streaming logic (and in general have granular properties)?

Reply:
thought about it but `SessionSettings` don't include the optimizer rules which are added dynamically.

Reply:
I think it worth adding a comment that Map can have dynamically optimizer rules